backend/fms_core/migrations/0065_v4_12_0.py

```diff
@@ -1,43 +1,64 @@
+            {"display_name": "Sample QC Type", "sheet_name": "SampleQC", "column_name": "Sample Kind", "value": "RNA"}]
+        }
+    ]
 
     with reversion.create_revision(manage_manually=True):
         admin_user = User.objects.get(username=ADMIN_USERNAME)
-        
-        reversion.set_comment(f"Set '{PROPERTY_TYPE_NAME}' property types as optional.")
+        reversion.set_comment(f"Create Sample QC DNA and RNA step and transfer existing Sample QC data to their respective new step.")
         reversion.set_user(admin_user)
-
-        for protocol_name in PROTOCOL_NAMES:
-            object_id = Protocol.objects.get(name=protocol_name).id
-            property_type = PropertyType.objects.get(name=PROPERTY_TYPE_NAME, object_id=object_id, content_type=protocol_content_type)
-            property_type.is_optional = True
-            property_type.save()
-            reversion.add_to_revision(property_type)
+        for step_info in STEPS:
+            protocol = Protocol.objects.get(name=step_info["protocol_name"])
+            step = Step.objects.create(name=step_info["name"],
+                                       protocol=protocol,
+                                       created_by_id=admin_user.id,
+                                       updated_by_id=admin_user.id)
+            reversion.add_to_revision(step)
+            for specification in step_info["specifications"]:
+                step_specification = StepSpecification.objects.create(display_name=specification["display_name"],
+                                                                      sheet_name=specification["sheet_name"],
+                                                                      column_name=specification["column_name"],
+                                                                      value=specification["value"],
+                                                                      step=step,
+                                                                      created_by_id=admin_user.id,
+                                                                      updated_by_id=admin_user.id)
+                reversion.add_to_revision(step_specification)
+        oldStep = Step.objects.get(name="Sample QC")
+        if oldStep:
+            sns = SampleNextStep.objects.filter(step__id=oldStep.id)
+            for sampleNextStep in sns:
+                sampleNextStepByStudy = SampleNextStepByStudy.objects.get(sample_next_step__id=sampleNextStep.id)
+                stepOrder = StepOrder.objects.get(id=sampleNextStepByStudy.step_order.id)
+                if sampleNextStep.sample.derived_samples.first().sample_kind.name == "RNA":
+                  newStep = Step.objects.get(name="Sample QC (RNA)")
+                if sampleNextStep.sample.derived_samples.first().sample_kind.name == "DNA":
+                  newStep = Step.objects.get(name="Sample QC (DNA)")
+                sampleNextStep.update(step=newStep)
```

Comment:
I recall issues with update and reversion. Make sure that once the migration is executed a new version of the sampleNextStep is stored for each update(). If you want to save on tests you can use an assignment to the attribute of the object and save the object instead.